Add 'Flip' Mod to osu!taiko

PermafrostFreeze opened the pull request:
This mod mirrors the playfield in a fashion similar to how catch's 'floating fruit' mod does.
The mod simply inverts the x axis and reassigns the anchor point, and there appears to be no conflicts with any mods (HD, FL, CL).



https://github.com/user-attachments/assets/ccdef390-a229-4901-a14a-e926055138ba

PermafrostFreeze committed:
add taiko mod

Shavixinio commented on osu.Game.Rulesets.Taiko/TaikoRuleset.cs:
```diff
@@ -166,7 +166,8 @@ public override IEnumerable<Mod> GetModsFor(ModType type)
                         new TaikoModMuted(),
-                        new ModAdaptiveSpeed()
+                        new ModAdaptiveSpeed(),
+                        new TaikoModMirror()
```
This should probably be a conversion mod type

Freeze7x replied:
I considered putting it in there, but it doesn’t impact the beatmap or the audio. And since the Floating Fruits mod is under the Fun section and works nearly the same as this mod, I thought it would be best to put it under the same section.

bdach replied:
It should not be conversion because conversion changes the map. (Alternate / Single Tap in osu! kinda don't do it, not sure they should continue to be conversion, but that's not a conversation for this PR.)

bdach commented on osu.Game.Rulesets.Taiko/Mods/TaikoModMirror.cs:
```diff
@@ -0,0 +1,25 @@
+{
+    public class TaikoModMirror : Mod, IApplicableToDrawableRuleset<TaikoHitObject>
+    {
+        public override string Name => "Mirror";
+        public override string Acronym => "MR";
```
I'm not sure I would want to name this "mirror" though. In osu! and mania the mirror mod has a strong connotation with changing your inputs. This does nothing of that sort, it only affects how you read.

To illustrate why this is bad: we have behaviours like changing the ruleset preserves mods by acronym, which seems not helpful or intentional in this instance.

I'm drawing a blank on better names at this point though.

bdach commented on osu.Game.Rulesets.Taiko/Mods/TaikoModMirror.cs:
```diff
@@ -0,0 +1,25 @@
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<TaikoHitObject> drawableRuleset)
+        {
+            drawableRuleset.Scale = new Vector2(-1, 1);
```
This is a quick and simple implementation but it also has several issues.

Judgement text mirrored:
![Screenshot 2025-05-07 at 10 01 25](https://github.com/user-attachments/assets/2b45225d-a115-4d93-98c0-ec8a58ae8e4c)

Clutter in top right on skins with mascots:
![Screenshot 2025-05-07 at 10 03 37](https://github.com/user-attachments/assets/b8df585d-4d0a-4749-a5cf-948ffe43f0cc)

I'm not sure this can be resolved amicably without some rather invasive changes, and thus my current stance is that we probably shouldn't get this mod in unless there's somehow overwhelming support for it.

PermafrostFreeze committed:
Rename to Flip

PermafrostFreeze committed:
also rename class to flip

PermafrostFreeze committed:
and rename the script to flip as well

I keep forgetting to rename everything in one go, these constant commits were not on purpose.

Shavixinio committed:
Reword the description

PermafrostFreeze committed:
Merge pull request #3 from Shavixinio/patch-6

Reword the description of Taiko Flip mod
